src/drt/src/db/obj/frInst.h

```diff
@@ -94,6 +95,15 @@ class frInst : public frRef
   void setOrigin(const Point& tmpPoint) override { xform_.setOffset(tmpPoint); }
   dbTransform getTransform() const override { return xform_; }
   void setTransform(const dbTransform& xformIn) override { xform_ = xformIn; }
+  void setDBInst(odb::dbInst* db_inst_in) { db_inst_ = db_inst_in; }
```

Comment:
make it a parameter in the constructor. There should never be an frInst that is not linked to dbInst